crates/circuit/src/circuit_data.rs

```diff
@@ -96,7 +96,7 @@ pub struct CircuitData {
     clbits: BitData<Clbit>,
     param_table: ParamTable,
     #[pyo3(get)]
-    global_phase: Param,
+    pub global_phase: Param,
```

Comment:
At least from #12794 onwards (and probably now anyway), this needs to be a private attribute - it's not safe to directly set it in general, because you always have to remember to untrack any `Parameter` uses within the global phase and track any new ones (not there are any here).

In #12794 there's a setter `CircuitData::set_global_phase` for it.

Reply:
Lets wait for #12794 then, this PR isn't critical

Reply:
Done in: https://github.com/Qiskit/qiskit/pull/12809/commits/d5c96937d55687a57baeabc52d23fe12c0631add